parse_str_test.go

```diff
@@ -4,27 +4,45 @@ import (
 	"fmt"
 	"reflect"
 	"testing"
+
+	"github.com/elliotchance/orderedmap/v2"
 )
 
+type dict map[any]any
+
+func convertOrderedMapToMap(omap orderedmap.OrderedMap[any, any]) dict {
```

Comment:
The inherent complexity and the unique structure of `OrderedMap` make it challenging to directly compare the outcomes tests against expected results. Therefore, I've added a new function, `convertOrderedMapToMap`, to facilitate comparison between `OrderedMap` types and standard Go maps in our tests.